### PR TITLE
Fix: Removed GPL code from FilesLauncher

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Description

I found the codes licensed under the GPL v3, though this whole project is licensed MIT license.

### Motivation and Context

GPL code cannot be existed in Files.

NOTE CONTRIBUTORS: Even if you want to revert changes, you definitely should not. Please find another way.

### Tasks

- [x] Remove codes that reference GPL code.
- [x] Remove GPL code.

### Validation

- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

**Comment by 0x5bfa** [comment]
@yair100 @gave92 How do you think of this?

File: Files.sln
Comment (yaira2) [inline Files.sln:5]: ?

File: src/Files.OpenDialog/FilesLauncher/ServiceProviderBase.cpp
Comment (hishitetsu) [inline src/Files.OpenDialog/FilesLauncher/ServiceProviderBase.cpp:2]: Is this work in progress? Even if you change the headers, it cannot be licensed under anything other than GPL as long as the original code remains. If you want to remove the GPL code, you must completely rewrite or delete the code.

Comment (onein528) [reply src/Files.OpenDialog/FilesLauncher/ServiceProviderBase.cpp:2]: I know and I discussed everything about licensing with Files owner. I changed this to all files. and As you can see PR header comment, I have tasks and will remove those files.

Comment (hishitetsu) [reply src/Files.OpenDialog/FilesLauncher/ServiceProviderBase.cpp]: OK, I understand that they will eventually be removed.

**Comment by hishitetsu** [comment]
Isn't there any functional regression to this change?

**Comment by onein528** [comment]
I'm not sure. and so I'm asking Yair and Gave, who is original creator.

**Comment by hishitetsu** [comment]
I have tested it. It seems to work well except highlighting a file as noted [here](https://github.com/files-community/Files/pull/11483#issuecomment-1445760512).
I think this level of regression would be acceptable.

**Comment by yaira2** [comment]
Can you fix the merge conflict?

File: src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp
Comment (hishitetsu) [inline src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:136]: Is it appropriate to delete the comment?

File: src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp
Comment (hishitetsu) [inline src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:1]:
```suggestion
// Copyright (c) 2023 Files Community
```

File: src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp
Comment (hishitetsu) [inline src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp:1]:
```suggestion
// Copyright (c) 2023 Files Community
```

File: src/Files.OpenDialog/FilesLauncher/OpenInFolder.h
Comment (hishitetsu) [inline src/Files.OpenDialog/FilesLauncher/OpenInFolder.h:1]:
```suggestion
// Copyright (c) 2023 Files Community
```

File: src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp
Comment (hishitetsu) [inline src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp:94]: Delete the commented out code.

File: src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp
Comment (onein528) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:136]: Absolutely, YES. Need C++ dev for review.

File: src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp
Comment (onein528) [reply src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp:94]: Finally, yes. still remain.

File: src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp
Comment (0x5bfa) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp]: @yair100 This is what I’m wondering always. Why Community has being licensed? Do you mean we all have attribution to Files?

Comment (onein528) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:1]: I’m going to leave it as it is. Just app name.

**Comment by onein528** [comment]
Done here. requesting a new review.

File: src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp
Comment (onein528) [reply src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp:94]: Probably shouldn't remove it in order to revert this only if the critical issue appears. But I'm not sure.

Comment (hishitetsu) [reply src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp:94]: I don't think it is necessary to leave the commented out code because deleted code can be easily restored from the commit history.

File: src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp
Comment (hishitetsu) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:136]: OK. You're right.

File: src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp
Comment (hishitetsu) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp]: Let @yair100 decide. I think the copyright notation should be aligned with the LICENSE file.

**Comment by hishitetsu** [comment]
@onein528 Could you merge the main branch and build FilesLauncher again? The file FilesLauncher.exe.sha256 should be generated.

**Comment by onein528** [comment]
I'll be able to work on that the next week

File: src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp
Comment (hishitetsu) [inline src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp:74]: This function has same name and parameters as Explorer++ and has a similar code, so we should remove this as well.
I believe `OpenInFolder` class is only used to make a particular file selected at startup and this file can be deleted completely if we give up this feature.

**Comment by onein528** [comment]
@hishitetsu What did you say before on Discord? I remember a bit you were talking about unneeded codes.

File: src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp
Comment (onein528) [reply src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp:74]: That doesn't matter even though it has same name and arguments.

Comment (hishitetsu) [reply src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp:74]: Is it "based on" as defined in the GPL license?

> To “modify” a work means to copy from or adapt all or part of the work in a fashion requiring copyright permission, other than the making of an exact copy. The resulting work is called a “modified version” of the earlier work or a work “based on” the earlier work.

Comment (onein528) [reply src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp:74]: - Same function name
- Same function arguemnt
- Different namespace
- Different codes contain

Comment (hishitetsu) [reply src/Files.OpenDialog/FilesLauncher/OpenInFolder.cpp]: Well, it's a delicate point. I'm no expert either, so I'll leave it to your judgment.

**Comment by onein528** [comment]
How can I fix this conflict

**Comment by hishitetsu** [comment]
> How can I fix this conflict

FilesystemOperationDialog.xaml is modified in #11557, so you can imported the one in the main branch.

**Comment by hishitetsu** [comment]
If you say about FilesLauncher.exe, you can import the one in your branch, merge the upstream, then build FilesLauncher again.

**Comment by 0x5bfa** [comment]
@hishitetsu Thank you:)

**FYI**
To avoid conflict, I reverted changes in the exe file and I'm going to open another PR to update the assembly after this merged.

Finally! @yair100 Ready for review.

**Review by hishitetsu — APPROVED** [review]
LGTM

**Pull request merged**
